tasks/agent.py

```diff
@@ -129,6 +129,15 @@ def build(
         python_runtimes=python_runtimes,
     )
 
+    if sys.platform == 'darwin' and platform.processor() == 'arm64':
```

Comment:
```suggestion
    if sys.platform == 'darwin' and platform.machine() == 'arm64':
```

For whatever reason my MBP M1 shows this:

```
% /usr/bin/python3
Python 3.8.9 (default, Oct 26 2021, 07:25:53)
[Clang 13.0.0 (clang-1300.0.29.30)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.uname()
uname_result(system='Darwin', node='MWR542RXK7', release='21.4.0', version='Darwin Kernel Version 21.4.0: Mon Feb 21 20:35:58 PST 2022; root:xnu-8020.101.4~2/RELEASE_ARM64_T6000', machine='arm64', processor='arm')
>>> platform.machine()
'arm64'
>>> platform.processor()
'arm'
>>>
```

Additional suggestion - Have this automaticlally skip the `datadog-agent-three` / `datadog-agent-two` task, as this only makes go cross compile, but cmake for `rtloader` / `datadog-agent-three` / `datadog-agent-two` still builds for the native host's CPU arcitecure, otherwise you get something like this:

```
It seems that you're running a Mac M1. Cross-compiling for amd64 then.
# github.com/DataDog/datadog-agent/pkg/collector/python
ld: warning: ignoring file /Users/aegelhofer/go/src/github.com/DataDog/datadog-agent/dev/lib/libdatadog-agent-rtloader.dylib, building for macOS-x86_64 but attempting to link with file built for macOS-arm64
Undefined symbols for architecture x86_64:
```

Which you can work around by running `invoke agent.build --build-exclude=systemd,python`.

Thanks for this though, this unblocks me from developing on a MBP M1 in https://github.com/DataDog/datadog-agent/issues/11677

Reply:
Great suggestion, thanks ! It might be because we have a different MBP M1. Mine ran:

```
Python 3.9.12 (main, Mar 26 2022, 15:44:31)
[Clang 13.1.6 (clang-1316.0.21.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> import platform
>>> platform.uname()
uname_result(system='Darwin', node='xxxx', release='21.4.0', version='Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000', machine='arm64')
>>> platform.machine()
'arm64'
>>> platform.processor()
'arm64'
>>> platform.platform()
'macOS-12.3.1-arm64-64bit'
```

As for the task skipping suggestion, this PR does not fix building the Python-related parts of the code, so you either have to exclude `python` or build the `iot` flavor. We will work on this in the future, thanks !

Reply:
Your "python3" might be something installed with homebrew, that was compiled with "more correct" values - Mine (`/usr/bin/python3`) is the "stock" python3 interpreter that comes from Apple.